### PR TITLE
Create free functions for some checks in java trace validation

### PR DESCRIPTION
**Pull request by jeannielynnmoulton**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 66909b0d3c61b68907146172adb7403f574e70be).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117314130

File: jbmc/src/java_bytecode/java_trace_validation.cpp [hunk 1]
```diff
@@ -115,6 +115,18 @@ bool check_address_structure(const address_of_exprt &address)
...
+bool check_constant_structure(const constant_exprt &constant_expr)
+{
+  if(constant_expr.has_operands())
+  {
+    const auto &operand = skip_typecast(constant_expr.operands()[0]);
+    return operand.id() == ID_constant || operand.id() == ID_address_of ||
+           operand.id() == ID_plus;
+  }
+  else
+    return !constant_expr.get_value().empty();
```
Comment (LAJW) [inline jbmc/src/java_bytecode/java_trace_validation.cpp]: ⛏ Unnecessary else before return 😁

File: jbmc/src/java_bytecode/java_trace_validation.cpp [hunk 2]
```diff
@@ -115,6 +115,18 @@ bool check_address_structure(const address_of_exprt &address)
   return symbol_expr && check_symbol_structure(*symbol_expr);
 }
 
+bool check_constant_structure(const constant_exprt &constant_expr)
+{
+  if(constant_expr.has_operands())
+  {
+    const auto &operand = skip_typecast(constant_expr.operands()[0]);
+    return operand.id() == ID_constant || operand.id() == ID_address_of ||
+           operand.id() == ID_plus;
```
Comment (LAJW) [inline jbmc/src/java_bytecode/java_trace_validation.cpp]: ⛏ I don't mind, but people will complain this isn't done with `can_cast_expr<plus_exprt>`, etc.

**Review by LAJW — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 53a1c6c1fb65615ecdda443867f58c38a87f76a9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117510344

**Pull request merged**